doc/rm/device_interface_functions.md

```diff
@@ -0,0 +1,167 @@
+Each hardware IP block also contains a "programmers' guide" section, which tells programmers how to use the hardware interfaces correctly, using prose descriptions and relevant diagrams.
+The prose descriptions should include references to relevant DIF functions.
+
+Programmers' guides **should** primarily refer to the [Software API Documentation]({{< relref "/sw#opentitan-software-api-documentation" >}}).
+Programmers' guides **should not** contain standalone examples that do not match how we implement DIFs and can become out of date.
+The software API documentation includes full source code of the respective DIFs, should extra clarity be needed to explain how a hardware interface should be used.
+
+The programmers' guide must also include a list of DIF functions which relate to it, along with a brief description of each, and a link to further API documentation.
+
+## Source Layout
+
+DIFs **must** live in the `sw/device/lib/dif` folder.
+They **must** comprise at least one header unique to the hardware in question, and **may** contain more than one C file providing the implementation, if the complexity of the implementation warrants it (this is subjective).
+Any files for a given IP should have a filename starting `dif_<IP name>`.
+
+Verification-specific extensions and logic **must not** live in `sw/device/lib/dif`.
+This code must be placed in the directory belonging to the verification domain which the extension is applicable to.
+Verification-specific extensions must not be built and included in production firmware builds.
+
+## Testing
+
+Tests **must** live in the `sw/device/tests/dif` folder.
+
+DIFs **must** have unit tests, which cover all their specification-required functionality.
+These unit tests use a mocking system which allows the test to control all hardware reads and writes, without requiring complex management of the hardware.
+These unit tests are written using [googletest](https://github.com/google/googletest), and may be run on a non-RISC-V host.
+
+DIFs **should** also have other tests, including standalone C sanity tests---which can quickly diagnose major issues between the DIF and the hardware but are not required to test all device functionality.
+
+DIFs are also being used by DV for chip-level tests, which should help catch any issues with the DIFs not corresponding to the hardware implementation exactly.
+These tests may not live in `sw/device/tests/dif`.
+
+## DIF Development Stages
+
+As part of the DIF Standardisation process, we’ve decided to establish more concrete DIF lifecycle stages.
+These are documented with the [Development Stages]({{< relref "doc/project/development_stages.md" >}}).
+
+In the hardware world, there are checklists for establishing that each stage is complete and the next stage can be moved to.
+
+In Software, we don’t have to follow the same stability guarantees, because software is much more modifiable.
```

Comment:
Nit: "Software" seems randomly capitalised

Reply:
Done, and fixed the non-ascii apostrophe.